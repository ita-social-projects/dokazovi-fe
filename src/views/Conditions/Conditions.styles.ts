import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles(
  (theme: Theme) => ({
    root: {},
  }),
  { name: 'ConditionsView' },
);
