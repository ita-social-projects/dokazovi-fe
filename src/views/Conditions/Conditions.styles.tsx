import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  (theme) => ({
    container: {
      [theme.breakpoints.down('xs')]: {
        padding: 0,
      },
    },
  }),
  { name: 'Conditions' },
);
