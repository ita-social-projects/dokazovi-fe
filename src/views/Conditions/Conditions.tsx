/* eslint-disable react/jsx-curly-brace-presence */
import React from 'react';
import { Container, Box } from '@material-ui/core';
import ConditionNav from '../../components/ConditionNav';
import ContentSection from '../../components/ContentSection';

export default function Conditions(): JSX.Element {
  return (
    <Container>
      <Box>
        <ConditionNav />
        <ContentSection type={'about'} />
        <ContentSection type={'rules'} />
        <ContentSection type={'contacts'} />
      </Box>
    </Container>
  );
}
