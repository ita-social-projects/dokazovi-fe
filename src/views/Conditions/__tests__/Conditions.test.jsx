/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-return */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Conditions from '../Conditions';

const store = {
  headerVisibility: false,
  info: {
    1: {
      loading: 'succeeded',
      error: null,
      data: {
        id: 1,
        title: 'Про платформу',
        text:
          '<p>Доказові — платформа з перевіреною і надійною інформацією про COVID-19, вакцинацію та інші важливі теми, пов’язані з медициною.</p><p>На платформі публікуються авторські матеріали лікарів та експертів, інформація від ЮНІСЕФ, а також переклади міжнародних матеріалів та науково-популярних статей.</p><p>Автори платформи Доказові — це люди, які постійно навчаються, використовують дані з наукових джерел і належать до авторитетних медичних спільнот. Вони постійно підвищують рівень знань, щоб ефективно лікувати, розповідати і пояснювати.</p><p>Принципи, за якими працює платформа Доказові:</p><ul><li>доказова медицина: на порталі оприлюднюються матеріали, які відповідають принципу доказової медицини</li><li>відбір інформації експертами: всю інформацію для порталу відбирають, готують і верифікують авторитетні й надійні лікарі, експерти, фактчекери</li><li>міжнародні рекомендації: відомості, подані на порталі, відповідають рекомендаціям ВООЗ, Центрів контролю за захворюваннями США і сайту системи охорони здоров’я Великобританії</li><li>інформація, якою варто дилітись: закликаємо поширювати інформацію, оприлюднену на платформі, а також використовувати її з освітніми цілями (докладніше — у <a href="http://localhost:3000/conditions/rules" rel="noopener noreferrer" target="_blank">Правилах користування</a>)</li></ul><p>Матеріали сайту будуть корисними сімейним лікарям та медикам вузьких спеціальностей, науковцям, які працюють з темами COVID-19 і вакцинації, журналістам, які пишуть про них, батькам та всім, хто хоче знати більше на теми, про які пише платформа Доказові.</p><p>Платформу розробили та запустили у 2021 році Дитячий фонд ООН (ЮНІСЕФ) в Україні та SoftServe за підтримки Агентства США з міжнародного розвитку (USAID).</p>',
      },
    },
    2: {
      loading: 'succeeded',
      error: null,
      data: {
        id: 2,
        title: 'Правила використання',
        text:
          '<p>Платформа Доказові заохочує використовувати, публікувати і розповсюджувати оприлюднені на ній матеріали. Водночас використання матеріалів можливе за умов, викладених у цих правилах.</p><p>Матеріали, оприлюднені на платформі Доказові, створені для цієї платформи, якщо в самій публікації не вказано інше джерело (наприклад, що це допис експерта у соцмережах або відео організації).</p><p>Матеріали платформи дозволено використовувати:</p><p>в інших медіа –лише за умови посилання на платформу Доказові:</p><ul><li>для діджитал-медіа – гіперпосилання на dokazovi.info не нижче першого абзацу</li><li>для телебачення і радіо – згадка в синхроні сюжету/передачі</li><li>для особистих цілей: ви можете завантажити або скопіювати вміст сайту, інші компоненти та елементи, що відображаються на сайті</li><li>для освітнього процесу: як джерела для матеріалів занять, завдання тощо</li></ul><p>Під час використання матеріалів у будь-якому випадку не можна змінювати текст і візуальні матеріали, щоб запобігти тому, що сенс матеріалів може бути розтлумачено невірно, виділено з контексту або викривлено.</p><p>В будь-якому випадку під час використання матеріалів необхідно завжди вказувати його автора/авторів.</p><p>Використовувати матеріали платформи з метою отримання комерційної вигоди заборонено.</p><p><br></p><p>Переклади матеріалів видань ____, ____, _____ здійснено на умовах ексклюзивної угоди між ЮНІСЕФ в Україні та цими виданнями. УМОВИ ПЕРЕДРУКУ ЦИХ МАТЕРІАЛІВ.</p><p><em>Тут ще буде доповнення по тексту, поки за оцінками – до 1000 символів.</em></p>',
      },
    },
    3: {
      loading: 'succeeded',
      error: null,
      data: {
        id: 3,
        title: 'Контакти',
        text:
          '<p>Хочете стати автором платформи?</p><p>Бажаєте співпрацювати?</p><p>Маєте запитання?</p><p>Пишіть нам на contact@dokazovi.info</p>',
      },
    },
    isFetchedAll: true,
  },
};

jest.mock('react-redux', () => ({
  useSelector: (cb) => {
    return cb(store);
  },
  useDispatch: () => jest.fn(),
}));

describe('Conditions component tests', () => {
  it('should render Conditions component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Conditions />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render main blocks', () => {
    render(
      <MemoryRouter>
        <Conditions />
      </MemoryRouter>,
    );

    expect(screen.getAllByText('Про платформу')[1]).toBeInTheDocument();
    expect(screen.getAllByText('Правила використання')[1]).toBeInTheDocument();
    expect(screen.getAllByText('Контакти')[1]).toBeInTheDocument();
  });
});
