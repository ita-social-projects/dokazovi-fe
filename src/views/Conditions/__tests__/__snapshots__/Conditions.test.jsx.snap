// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditions component tests should render Conditions component 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root Conditions-container-1 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <ul
        class="
      ConditionsNav-root-3 
      
    "
      >
        <li>
          <a
            aria-current="page"
            class="ConditionsNav-link-5"
            data-testid="about"
            href="/#about"
          >
            Про платформу
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="ConditionsNav-link-5"
            data-testid="rules"
            href="/#rules"
          >
            Правила використання
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="ConditionsNav-link-5"
            data-testid="contacts"
            href="/#contacts"
          >
            Контакти
          </a>
        </li>
      </ul>
      <div
        class="ContentSection-section-7"
        id="about"
      >
        <div
          class="ContentSection-wrap-8"
        >
          <h2
            class="MuiTypography-root ContentSection-title-9 MuiTypography-h2"
          >
            Про платформу
          </h2>
        </div>
        <div
          class="ContentSection-content-11"
        >
          <p>
            Доказові — платформа з перевіреною і надійною інформацією про COVID-19, вакцинацію та інші важливі теми, пов’язані з медициною.
          </p>
          <p>
            На платформі публікуються авторські матеріали лікарів та експертів, інформація від ЮНІСЕФ, а також переклади міжнародних матеріалів та науково-популярних статей.
          </p>
          <p>
            Автори платформи Доказові — це люди, які постійно навчаються, використовують дані з наукових джерел і належать до авторитетних медичних спільнот. Вони постійно підвищують рівень знань, щоб ефективно лікувати, розповідати і пояснювати.
          </p>
          <p>
            Принципи, за якими працює платформа Доказові:
          </p>
          <ul>
            <li>
              доказова медицина: на порталі оприлюднюються матеріали, які відповідають принципу доказової медицини
            </li>
            <li>
              відбір інформації експертами: всю інформацію для порталу відбирають, готують і верифікують авторитетні й надійні лікарі, експерти, фактчекери
            </li>
            <li>
              міжнародні рекомендації: відомості, подані на порталі, відповідають рекомендаціям ВООЗ, Центрів контролю за захворюваннями США і сайту системи охорони здоров’я Великобританії
            </li>
            <li>
              інформація, якою варто дилітись: закликаємо поширювати інформацію, оприлюднену на платформі, а також використовувати її з освітніми цілями (докладніше — у 
              <a
                href="http://localhost:3000/conditions/rules"
                rel="noopener noreferrer"
                target="_blank"
              >
                Правилах користування
              </a>
              )
            </li>
          </ul>
          <p>
            Матеріали сайту будуть корисними сімейним лікарям та медикам вузьких спеціальностей, науковцям, які працюють з темами COVID-19 і вакцинації, журналістам, які пишуть про них, батькам та всім, хто хоче знати більше на теми, про які пише платформа Доказові.
          </p>
          <p>
            Платформу розробили та запустили у 2021 році Дитячий фонд ООН (ЮНІСЕФ) в Україні та SoftServe за підтримки Агентства США з міжнародного розвитку (USAID).
          </p>
        </div>
      </div>
      <div
        class="ContentSection-section-7"
        id="rules"
      >
        <div
          class="ContentSection-wrap-8"
        >
          <h2
            class="MuiTypography-root ContentSection-title-9 MuiTypography-h2"
          >
            Правила використання
          </h2>
        </div>
        <div
          class="ContentSection-content-11"
        >
          <p>
            Платформа Доказові заохочує використовувати, публікувати і розповсюджувати оприлюднені на ній матеріали. Водночас використання матеріалів можливе за умов, викладених у цих правилах.
          </p>
          <p>
            Матеріали, оприлюднені на платформі Доказові, створені для цієї платформи, якщо в самій публікації не вказано інше джерело (наприклад, що це допис експерта у соцмережах або відео організації).
          </p>
          <p>
            Матеріали платформи дозволено використовувати:
          </p>
          <p>
            в інших медіа –лише за умови посилання на платформу Доказові:
          </p>
          <ul>
            <li>
              для діджитал-медіа – гіперпосилання на dokazovi.info не нижче першого абзацу
            </li>
            <li>
              для телебачення і радіо – згадка в синхроні сюжету/передачі
            </li>
            <li>
              для особистих цілей: ви можете завантажити або скопіювати вміст сайту, інші компоненти та елементи, що відображаються на сайті
            </li>
            <li>
              для освітнього процесу: як джерела для матеріалів занять, завдання тощо
            </li>
          </ul>
          <p>
            Під час використання матеріалів у будь-якому випадку не можна змінювати текст і візуальні матеріали, щоб запобігти тому, що сенс матеріалів може бути розтлумачено невірно, виділено з контексту або викривлено.
          </p>
          <p>
            В будь-якому випадку під час використання матеріалів необхідно завжди вказувати його автора/авторів.
          </p>
          <p>
            Використовувати матеріали платформи з метою отримання комерційної вигоди заборонено.
          </p>
          <p>
            <br />
          </p>
          <p>
            Переклади матеріалів видань ____, ____, _____ здійснено на умовах ексклюзивної угоди між ЮНІСЕФ в Україні та цими виданнями. УМОВИ ПЕРЕДРУКУ ЦИХ МАТЕРІАЛІВ.
          </p>
          <p>
            <em>
              Тут ще буде доповнення по тексту, поки за оцінками – до 1000 символів.
            </em>
          </p>
        </div>
      </div>
      <div
        class="ContentSection-section-7"
        id="contacts"
      >
        <div
          class="ContentSection-wrap-8"
        >
          <h2
            class="MuiTypography-root ContentSection-title-9 MuiTypography-h2"
          >
            Контакти
          </h2>
        </div>
        <div
          class="ContentSection-content-11"
        >
          <p>
            Хочете стати автором платформи?
          </p>
          <p>
            Бажаєте співпрацювати?
          </p>
          <p>
            Маєте запитання?
          </p>
          <p>
            Пишіть нам на contact@dokazovi.info
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
